src/devices/bus/heathzenith/h8/cpu8080.cpp

```diff
@@ -0,0 +1,269 @@
+
+
+
+  Onboard Jumpers (none are configurable yet)
+
+  Label    Default     Description (if jumper is closed)
+  -------------------------------------------------------------------------
+  B1-B2     Open       Allow /INT1 (INT 10) from the BH Bus (pin 8).
+
+  C1-C2     Open       Allow /INT2 (Int 20) from the BH Bus (pin 9).
+
+  E1-E2     Close      Allow inverted signal from J jumper to HOLD/HLDA on BH Bus (pin 25)
+
+  F1-F2     Close      Allow /HOLD from the BH Bus (pin 27) to CPU Hold line.
+
+  H1-H3     Open       Tie Ready-in from IC212 Generator to CPU Wait line
+  H2-H3     Close      Tie Ready-in from IC212 Generator to CPU Ready line
+
+  J1-J2     Close      CPU HLDA out through flip-flop, use /Q output to E jumper
+  J2-J3     Open       CPU HLDA out through flip-flop, use Q output to E jumper
+
+  K1-K2     Close      Affects Data-in/out buffers enabled signals based on /MEMR & /IOR signals
+  K1-K3     Open       Affects Data-in/out buffers enabled signals based on /MEMR & /IOR signals
+
+  L1-L2     Close      Tie CPU /INTA to interrupt buffer
+  L2-L3     Open       Tie CPU /INTA to +12V
+
+  P1-P2     Open       A10 to pin 19 of IC204 ROM
+  P2-P3     Close      R jumper to pin 19 of IC204 ROM
+
+  R1-R2     Close      +12V selected
+  R2-R3     Open       S jumper selection selected
+
+  S1-S2     Open       +5V selected
+  S2-S3     Close      GND selected
+
+  T1-T2     Close      -5V to pin 21 of IC204 ROM
+  T2-T3     Open       +5V to pin 21 of IC204 ROM
+
+  X1-X2     Open       Allow /ROM_Disable from BH bus (pin 46) to disable ROM
+
+  Z1-Z2     Close      Tie /A10 to ROM decoder select line
+  Z2-Z3     Open       Tie +5V to ROM decoder select line
+
+  RDYIN     Close      Allow RDYIN from BH Bus (pin 20) to IC212 Generator
+
+****************************************************************************/
+
+#include "emu.h"
+#include "cpu8080.h"
+
+#include "bus/heathzenith/intr_cntrl/intr_cntrl.h"
+#include "cpu/i8085/i8085.h"
+
+
+namespace {
+
+class h_8_cpu_8080_device : public device_t, public device_h8bus_p2_card_interface
+{
+public:
+
+	h_8_cpu_8080_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock = 0);
+
+	virtual void int1_w(int state) override;
+	virtual void int2_w(int state) override;
+	virtual void int3_w(int state) override;
+	virtual void int4_w(int state) override;
+	virtual void int5_w(int state) override;
+	virtual void int6_w(int state) override;
+	virtual void int7_w(int state) override;
+
+	virtual void p201_reset_w(int state) override;
+	virtual void p201_int1_w(int state) override;
+	virtual void p201_int2_w(int state) override;
+
+protected:
+
+	virtual const tiny_rom_entry *device_rom_region() const override ATTR_COLD;
+	virtual void device_start() override ATTR_COLD;
+	virtual void device_add_mconfig(machine_config &config) override ATTR_COLD;
+
+private:
+	// 2.048 MHz
+	static constexpr XTAL H8_CLOCK = XTAL(18'432'000) / 9;
```

Comment:
Please make this a local in the machine configuration function, as it’s only used in one place (it increases static data size if you make it static).

Also, assuming the CPU card drives the bus clock, consider having the CPU card set the bus clock on start so it can propagate to other cards.